tests/unit/test_charm.py

```diff
@@ -926,3 +926,186 @@ def test_given_service_is_running_when_on_remove_then_service_is_stopped(
+        generate_role_secret_id.return_value = "11111111"
+
+        mount_suffix = "dummy"
+        self.harness.update_relation_data(rel_id, app_name, {"mount_suffix": mount_suffix})
+
+        set_mount.assert_not_called()
+        set_vault_url.assert_not_called()
+        set_ca_certificate.assert_not_called()
+        set_unit_credentials.assert_not_called()
+
+    @patch("vault.Vault.read_role_secret")
+    @patch("vault.Vault.generate_role_secret_id")
+    @patch("vault.Vault.configure_approle")
+    @patch("vault.Vault.configure_kv_policy")
+    @patch("vault.Vault.configure_kv_mount")
+    @patch("vault.Vault.is_api_available")
+    @patch("vault.Vault.enable_approle_auth")
+    def test_given_unit_is_leader_when_related_unit_egress_is_updated_then_secret_content_is_updated(  # noqa: E501
+        self,
+        enable_approle_auth,
+        __,
+        ___,
+        ____,
+        configure_approle,
+        generate_role_secret_id,
+        read_role_secret,
+    ):
+        peer_relation_id = self._set_peer_relation()
+        self._set_initialization_secret_in_peer_relation(
+            relation_id=peer_relation_id,
+            root_token="root token content",
+            unseal_keys=["unseal_keys"],
+        )
+        self._set_certificate_secret_in_peer_relation(
+            relation_id=peer_relation_id,
+            certificate="certificate content",
+            private_key="private key content",
+            ca_certificate="ca certificate content",
+        )
+        (
+            app_name,
+            _,
+            _,
+            rel_id,
+            units,
+        ) = self.setup_vault_kv_relation(nb_units=3)
+
+        configure_approle.return_value = "12345678"
+        generate_role_secret_id.return_value = "11111111"
+
+        mount_suffix = "dummy"
+        self.harness.update_relation_data(rel_id, app_name, {"mount_suffix": mount_suffix})
+        enable_approle_auth.assert_called()
+        # choose an unit to update
+        unit = next(iter(units.keys()))
+        unit_name = unit.replace("/", "-")
+
+        # Mock read to actually return a comparable cidr_list
+        def mock_read_role_secret(role_name, _):
+            unit_name = "/".join(role_name.split("-")[-2:])
+            return {"cidr_list": [units[unit_name]]}
+
+        read_role_secret.side_effect = mock_read_role_secret
+        # get current role secret id from unit's secert
+        generate_role_secret_id.return_value = "22222222"
+        with patch("ops.Secret.set_content") as set_content:
+            self.harness.update_relation_data(rel_id, unit, {"egress_subnet": "10.20.20.240/32"})
+            assert set_content.call_count == 1
+            set_content.assert_has_calls(
+                [call({"role-id": "12345678", "role-secret-id": "22222222"})]
+            )
+
+        assert read_role_secret.call_count == 3
+        read_role_secret.has_calls([call("charm-" + unit_name, "11111111")])
```

Comment:
This `has_calls` statement has no effect, you could have anything in it and the test would pass. (in other words, change for "assert_has_calls")

Reply:
Removed in last commit, this was unnecessary.